### PR TITLE
add karmadactl get -h example

### LINKED ISSUE #896: Provide some example for `karmadactl get`
**What would you like to be added**:
There is only  a template that is not clear. What resource we can get?
```
[root@master67 karmada]# karmadactl get -h
Display one or many resources

Usage:
  karmadactl get [NAME | -l label | -n namespace]  [flags]

Flags:
```

**Why is this needed**:
There aren't cluster when I execute `karmadactl get cluster`. Is it a bug or not supported?
```
[root@master67 karmada]# kubectl get cluster
NAME       VERSION   MODE   READY   AGE
member68   v1.19.1   Push   True    3d5h
[root@master67 karmada]# karmadactl get cluster
[cluster] from server (NotFound)
```

### PR DESCRIPTION
**Pull request by learner0810**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #896

**Special notes for your reviewer**:
The current examples are shown in the picture below. If you need to add examples, please let me know

企业微信截图_b38a27c5-b253-4a98-9d2a-f48f01029ee6

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by lonelyCZ** [comment]
Thanks for doing that! @learner0810 

Could you run this example in your environment? 

I have a new cert problem. 
```
[root@master67 yaml]# karmadactl get clusters --kubeconfig=/root/.kube/config --karmada-context=karmada-apiserver
Error: conversion webhook for work.karmada.io/v1alpha2, Kind=ResourceBinding failed: Post "https://karmada-webhook.karmada-system.svc:443/convert?timeout=30s": x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "ca")
```

**Comment by learner0810** [comment]
@lonelyCZ All examples have been tested in my environment.
企业微信截图_71af2f09-a935-4e70-aebd-c6195e7f41a1

But the get clusters result is as follows
企业微信截图_1a27e41c-3d26-4983-afa8-a4917127f371

**Comment by lonelyCZ** [comment]
> @lonelyCZ All examples have been tested in my environment. 企业微信截图_71af2f09-a935-4e70-aebd-c6195e7f41a1

All right! Pointing out `--kubeconfig=$HOME/.kube/members.config` is a key. Should we add it to Example?

File: pkg/karmadactl/get.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 	"k8s.io/client-go/rest"
 	"k8s.io/kubectl/pkg/cmd/get"
 	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:31]: We'd better not need these dependencies.

**Comment by learner0810** [comment]
> > @lonelyCZ All examples have been tested in my environment. 企业微信截图_71af2f09-a935-4e70-aebd-c6195e7f41a1
> 
> All right! Pointing out `--kubeconfig=$HOME/.kube/members.config` is a key. Should we add it to Example?

I don't think it's necessary because everyone's config file name is different

File: pkg/karmadactl/get.go (hunk 1 above)
Comment (learner0810) [reply pkg/karmadactl/get.go]: done

**Comment by lonelyCZ** [comment]
> I don't think it's necessary because everyone's config file name is different

It is a problem. We could use a common name replacing it. Users may not know to use kubeconfig of member clusters, because default context of `karmadactl` usually is karmada-apiserver.

**Comment by learner0810** [comment]
> > I don't think it's necessary because everyone's config file name is different
> 
> It is a problem. We could use a common name replacing it. Users may not know to use kubeconfig of member clusters, because default context of `karmadactl` usually is karmada-apiserver.

**Comment by learner0810** [comment]
> > I don't think it's necessary because everyone's config file name is different
> 
> It is a problem. We could use a common name replacing it. Users may not know to use kubeconfig of member clusters, because default context of `karmadactl` usually is karmada-apiserver.

May I ask if that's what you mean
```
karmadactl get pods --kubeconfig=$cluster-config --karmada-context=$cluster-context

**Comment by lonelyCZ** [comment]
I just try it in karmada created by Kind, it is no problem. It doesn't seem necessary. Thanks @learner0810

**Comment by learner0810** [comment]
> I just try it in karmada created by Kind, it is no problem. It doesn't seem necessary. Thanks @learner0810

Ok, may I ask what needs to be modified in this PR

File: pkg/karmadactl/get.go [hunk 2]
```diff
@@ -51,6 +51,22 @@ var (
 	noPushModeMessage = "The karmadactl get command now only supports Push mode, [ %s ] are not push mode\n"
 	getShort          = `Display one or many resources`
 	defaultKubeConfig = filepath.Join(os.Getenv("HOME"), ".kube/config")
+
+	getExample = `
+	# List all pods in ps output format
+	karmadactl get pods
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:57]: We can't seem to use `karmadactl` directly. Because we also have `kubectl karmada`.

Comment (RainbowMango) [reply pkg/karmadactl/get.go:57]: Yeah, that's a thing.

```
# ./kubectl-karmada get --help 
Display one or many resources

Usage:
  karmada get [NAME | -l label | -n namespace]  [flags]

Examples:

	# List all pods in ps output format
	karmadactl get pods

	# List a single replicasets controller with specified NAME in ps output format
	karmadactl get replicasets nginx

	# List deployments in JSON output format, in the "v1" version of the "apps" API group
	karmadactl get deployments.v1.apps

	# List all replication controllers and services together in ps output format
	karmadactl get rs,services

	# List one or more resources by their type and names
	karmadactl get rs/nginx-cb87b6d88 service/kubernetes

```

**Comment by learner0810** [comment]
@lonelyCZ @RainbowMango The output of the example is now shown below，PTAL

企业微信截图_78604011-12ac-42c0-8625-b33091526705
企业微信截图_46ea272f-0fd9-4abe-8c1f-61092f874a19
企业微信截图_5505a1f1-46a1-4968-9ed4-2361aa3bf584

**Comment by lonelyCZ** [comment]
Looks good. We could unify format like other command that no retract.

```
[root@master67 karmada]# karmadactl unjoin -h
Unjoin removes the registration of a cluster from control plane.

Usage:
  karmadactl unjoin CLUSTER_NAME --cluster-kubeconfig=<KUBECONFIG> [flags]

Examples:

# Unjoin cluster from karmada control plane
karmadactl unjoin CLUSTER_NAME --cluster-kubeconfig=<KUBECONFIG>

# Unjoin cluster from karmada control plane with timeout
karmadactl unjoin CLUSTER_NAME --cluster-kubeconfig=<KUBECONFIG> --wait 2m

# Unjoin cluster from karmada control plane, if the agent was deployed in custom namespace
karmadactl unjoin CLUSTER_NAME --cluster-kubeconfig=<KUBECONFIG> --agent-namespace=<CUSTOM_NAMESPACE>
```

**Comment by RainbowMango** [comment]
We don't have the `--agent-namespace=` flag yet, right?

**Comment by learner0810** [comment]
@lonelyCZ done，PTAL
企业微信截图_1662a4d0-69ae-41c2-b5f5-91a1b3cc3778
企业微信截图_f15e5a96-6a96-408c-a60c-0b6a7191fb1a

**Comment by lonelyCZ** [comment]
> We don't have the `--agent-namespace=` flag yet, right?

Yes, this is just an example.

**Comment by RainbowMango** [comment]
looks good. What's the `ps` output format? by the way.

File: pkg/karmadactl/get.go [hunk 3]
```diff
@@ -54,14 +54,15 @@ var (
 )
 
 // NewCmdGet New get command
-func NewCmdGet(out io.Writer, karmadaConfig KarmadaConfig) *cobra.Command {
+func NewCmdGet(out io.Writer, karmadaConfig KarmadaConfig, cmdStr string) *cobra.Command {
```
Comment (RainbowMango) [inline pkg/karmadactl/get.go:57]:
```suggestion
func NewCmdGet(out io.Writer, karmadaConfig KarmadaConfig, parentCommand string) *cobra.Command {
```

File: pkg/karmadactl/get.go [hunk 4]
```diff
@@ -549,3 +550,23 @@ func Exists(path string) bool {
 	}
 	return true
 }
+
+// getExample get examples by cmd type
+func getExample(cmdStr string) string {
```
Comment (RainbowMango) [inline pkg/karmadactl/get.go:555]:
```suggestion
func getExample(parentCommand string) string {
```

File: pkg/karmadactl/karmadactl.go [hunk 5]
```diff
@@ -47,7 +47,7 @@ func NewKarmadaCtlCommand(out io.Writer, cmdUse, cmdStr string) *cobra.Command {
 	rootCmd.AddCommand(sharedcommand.NewCmdVersion(out, cmdStr))
 	rootCmd.AddCommand(NewCmdCordon(out, karmadaConfig, cmdStr))
 	rootCmd.AddCommand(NewCmdUncordon(out, karmadaConfig, cmdStr))
-	rootCmd.AddCommand(NewCmdGet(out, karmadaConfig))
+	rootCmd.AddCommand(NewCmdGet(out, karmadaConfig, cmdStr))
```
Comment (RainbowMango) [inline pkg/karmadactl/karmadactl.go:50]: I suggest rename this `cmdStr` to `parentCommand` as well.

**Review by RainbowMango — COMMENTED** [review]
LGTM. Just a suggestion.

File: pkg/karmadactl/karmadactl.go (hunk 5 above)
Comment (learner0810) [reply pkg/karmadactl/karmadactl.go]: Already modified. PTAL

**Comment by RainbowMango** [comment]
Thanks.
/lgtm
/approve

**Comment by learner0810** [comment]
> looks good. What's the `ps` output format? by the way.

I didn't find a specific definition of ps output format, but I saw a blog that said it was like the format of the linux ps command output.kubectl's also uses ps output format .

**Comment by RainbowMango** [comment]
Ok. I know we borrowed the concept from kubectl, but it'll be embarrassing if people ask the same question. :)

**Pull request merged**